synapse/config/server.py

```diff
@@ -540,6 +540,9 @@ class LimitRemoteRoomsConfig(object):
         if not isinstance(self.users_new_default_push_rules, list):
             raise ConfigError("'users_new_default_push_rules' must be a list")
 
+        # Turn the list into a set to improve lookup speed.
+        self.users_new_default_push_rules = set(self.users_new_default_push_rules)
```

Comment:
this means that `self.users_new_default_push_rules` changes type from list to set. could you just use a local variable rather than a class member for the `list` version?

Reply:
Oh yeah good shout.